[E2E] Re-enable custom mapping null test

nemanjaglumac opened the pull request:
This test was quarantined as flaky, but I found it to be actually wrong.

This PR fixes the test and unskips it.
The stress-test workflow confirms that it's not flaking.

- 1 x 20 ✅ https://github.com/metabase/metabase/actions/runs/6943906753
- 1 x 40 ✅ https://github.com/metabase/metabase/actions/runs/6943975267

nemanjaglumac committed:
Re-enable custom mapping null values test

nemanjaglumac committed:
Remove misleading comment

nemanjaglumac commented on e2e/test/scenarios/admin/datamodel/field.cy.spec.js:
```diff
@@ -16,7 +16,6 @@ import { SAMPLE_DATABASE } from "e2e/support/cypress_sample_database";
 
 const { ORDERS, ORDERS_ID } = SAMPLE_DATABASE;
 
-// [quarantine] - intermittently failing, possibly due to a "flickering" element (re-rendering)
```
Unrelated to this particular test.
Just a misleading comment.

nemanjaglumac commented on e2e/test/scenarios/admin/datamodel/field.cy.spec.js:
```diff
@@ -134,39 +133,42 @@ describe("scenarios > admin > datamodel > field", () => {
         dbId,
-        ({ number_with_nulls: { num }, number_with_nulls_ID }) =>
-          cy.visit(
-            `/admin/datamodel/database/${dbId}/schema/${SAMPLE_DB_SCHEMA_ID}/table/${number_with_nulls_ID}/field/${num}/general`,
```
Using `SAMPLE_DB_SCHEMA_ID` was wrong. SQLite has a different schema.

nemanjaglumac commented on e2e/test/scenarios/admin/datamodel/field.cy.spec.js:
```diff
@@ -134,39 +133,42 @@ describe("scenarios > admin > datamodel > field", () => {
+
       withDatabase(
         dbId,
-        ({ number_with_nulls: { num }, number_with_nulls_ID }) =>
```
These values were always undefined because they exist only as all caps fields.

WiNloSt commented on e2e/test/scenarios/admin/datamodel/field.cy.spec.js:
```diff
@@ -134,39 +133,42 @@ describe("scenarios > admin > datamodel > field", () => {
+            .click();
+          popover().findByText("Custom mapping").click();
+
+          cy.findByDisplayValue("null").clear().type(remappedNullValue).blur();
```
I think `.blur()` is unnecessary. When you click the save button in Cypress the `blur` event should be called anyway. And mostly I found we use `blur` when we're autosaving the field, which is not what we do in this case.

WiNloSt commented on e2e/test/scenarios/admin/datamodel/field.cy.spec.js:
```diff
@@ -134,39 +133,42 @@ describe("scenarios > admin > datamodel > field", () => {
+
+          cy.findByDisplayValue("null").clear().type(remappedNullValue).blur();
+          cy.button("Save").click();
+          cy.button("Saved!");
```
We should add an assertion to avoid confusion. Using `testing-library` to find an element without acting or asserting on it is discouraged.
```suggestion
          cy.button("Saved!").should("be.visible");
```

nemanjaglumac replied:
I'm actually in a group of people who prefer implicit assertions, but I hear you.
Good call.

Thank you <3

uladzimirdev replied:
➕, I'm fine with implicit assertions

WiNloSt replied:
@uladzimirdev I really couldn't find where the discussion was, but explicit assertions have no disadvantages.

It could be ambiguous what an author intends to do with the element because there could be multiple things you could do with an element:
1. Assert that it's visible, not visible, not in the DOM, is in the DOM, etc.
2. Click it
3. type in it (if it's an input)
etc.

And I wouldn't know if there's no explicit action on an element whether you wanted to assert it or wanted to click it but forgot to add the `.click()` or something else.

In short, I still think it's worth being explicit with this.

WiNloSt reviewed (approved):
I have some minor comments regarding the tests, but this works great!

nemanjaglumac committed:
Address review comments

nemanjaglumac commented:
> I have some minor comments regarding the tests, but this works great!

Thanks for the thorough review, as always.
Addressed the comments and ran another stress-test, just to make sure the test still passes without hiccups.
40/40 ✅ https://github.com/metabase/metabase/actions/runs/6954567753
